### PR TITLE
[WIP]Show notification when my session starts

### LINKED ISSUE #144: Show notification when my session starts
## Kind (Required)

- Improvement

 ## Overview (Required)

- Title says everything ⏲
- Probably we can use implementation of 2019.

 ## Links	

- https://github.com/DroidKaigi/conference-app-2019/pull/437

### PR DESCRIPTION
**Pull request by Daishin16**
## Issue
- close #144 

## Overview (Required)
- copied from last year
- fix package name, build.gradle, other library related
## Links
-

## Screenshot
Before | After
:--: | :--:
 |

### CODE REVIEW

File: feature/notification/src/main/java/io/github/droidkaigi/confsched2020/notification/worker/ManageTopicSubscriptionWorker.kt [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import com.google.firebase.iid.FirebaseInstanceId
+import com.google.firebase.messaging.FirebaseMessaging
+import io.github.droidkaigi.confsched2020.notification.Topic
+import io.github.droidkaigi.confsched2020.timber.error
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.tasks.await
+import timber.log.Timber
+import timber.log.debug
+
+class ManageTopicSubscriptionWorker(
```
Comment (takahirom) [inline feature/notification/src/main/java/io/github/droidkaigi/confsched2020/notification/worker/ManageTopicSubscriptionWorker.kt]: Thanks!
I think this is not a related code.
Class for receiving announcements from DroidKaigi.

File: feature/notification/src/main/java/io/github/droidkaigi/confsched2020/notification/service/FCMService.kt [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+import androidx.localbroadcastmanager.content.LocalBroadcastManager
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+import io.github.droidkaigi.confsched2020.notification.NotificationChannelInfo
+import io.github.droidkaigi.confsched2020.notification.NotificationUtil
+import io.github.droidkaigi.confsched2020.notification.createDefaultNotificationChannel
+import io.github.droidkaigi.confsched2020.timber.error
+import timber.log.Timber
+
+class FCMService : FirebaseMessagingService() {
```
Comment (takahirom) [inline feature/notification/src/main/java/io/github/droidkaigi/confsched2020/notification/service/FCMService.kt]: Thanks!
I think this is not a related code.
Class for receiving announcements from DroidKaigi.

File: feature/notification/src/main/java/io/github/droidkaigi/confsched2020/notification/worker/ManageTopicSubscriptionWorker.kt (hunk 1 above)
Comment (takahirom) [reply feature/notification/src/main/java/io/github/droidkaigi/confsched2020/notification/worker/ManageTopicSubscriptionWorker.kt]: I think most of the code to receive from firebase.
Can that part be removed once? 🙇

Comment (Daishin16) [reply feature/notification/src/main/java/io/github/droidkaigi/confsched2020/notification/worker/ManageTopicSubscriptionWorker.kt]: Thank you for your advice！
I haven't made a pull request so much ... I deleted something I thought was unnecessary and pushed it. Is the change reflected?

Comment (Daishin16) [reply feature/notification/src/main/java/io/github/droidkaigi/confsched2020/notification/worker/ManageTopicSubscriptionWorker.kt]: I noticed by email, do I need to delete the same for the woker and service classes?

Comment (takahirom) [reply feature/notification/src/main/java/io/github/droidkaigi/confsched2020/notification/worker/ManageTopicSubscriptionWorker.kt]: Yes. The task of receiving the announcement has been assigned to another person. 🙇 
https://github.com/DroidKaigi/conference-app-2020/issues/145

Comment (Daishin16) [reply feature/notification/src/main/java/io/github/droidkaigi/confsched2020/notification/worker/ManageTopicSubscriptionWorker.kt]: The fix is ​​complete, is this OK?

**Comment by takahirom** [comment]
Thanks!
Can you search "announce" and delete it ?🙇

**Comment by Daishin16** [comment]
announcement folder? Are these two objects in the Topic class?

**Comment by Daishin16** [comment]
Deletion complete

File: feature/notification/src/main/java/io/github/droidkaigi/confsched2020/notification/NotificationChannelInfo.kt [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+) {
+    DEFAULT(
+        "default_channel",
+        R.string.notification_channel_name_default
+    ),
+    FAVORITE_SESSION_START(
+        "favorite_session_start_channel",
+        R.string.notification_channel_name_start_favorite_session
+    ),
+    ANNOUNCEMENT(
```
Comment (takahirom) [inline feature/notification/src/main/java/io/github/droidkaigi/confsched2020/notification/NotificationChannelInfo.kt]: 🙏

File: feature/notification/src/main/res/values-ja/strings.xml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="notification_channel_name_default">その他</string>
+    <string name="notification_channel_name_start_favorite_session">マイセッションの開始</string>
+    <string name="notification_channel_name_announcement">運営からの広報</string>
```
Comment (takahirom) [inline feature/notification/src/main/res/values-ja/strings.xml]: 🙏

File: feature/notification/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<resources>
+    <string name="notification_channel_name_default">Not categorized</string>
+    <string name="notification_channel_name_start_favorite_session">start favorite session</string>
+    <string name="notification_channel_name_announcement">Announcement</string>
```
Comment (takahirom) [inline feature/notification/src/main/res/values/strings.xml]: 🙏

Comment (Daishin16) [reply feature/notification/src/main/res/values/strings.xml]: OK. Should I restore the deleted Topic class?

Comment (takahirom) [reply feature/notification/src/main/res/values/strings.xml]: You do not need to revert. 🙏 
I just want to remove announcement-related changes from this PR. 🙇

Comment (Daishin16) [reply feature/notification/src/main/res/values/strings.xml]: Deletion complete

File: feature/notification/src/main/AndroidManifest.xml [hunk 6]
```diff
@@ -0,0 +1,31 @@
+<manifest
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    package="io.github.droidkaigi.confsched2020.notification"
+    >
+
+    <uses-permission android:name="android.permission.FOREGROUND_SERVICE"/>
+    <application>
+        <service
+            android:name=".service.FCMService"
```
Comment (takahirom) [inline feature/notification/src/main/AndroidManifest.xml]: Sorry 🙇  Can you delete this??

File: feature/notification/build.gradle [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+    implementation Dep.Coil.coil
+
+    testImplementation project(':corecomponent:androidtestcomponent')
+    testImplementation Dep.Test.junit
+    testImplementation Dep.Test.kotlinTestAssertions
+    testImplementation Dep.MockK.jvm
+    androidTestImplementation Dep.Test.testRunner
+    androidTestImplementation Dep.Test.espressoCore
+    implementation 'androidx.localbroadcastmanager:localbroadcastmanager:1.0.0'
```
Comment (takahirom) [inline feature/notification/build.gradle]: Please delete this(now we are not using local broadcast manager) 🙏

File: feature/notification/build.gradle [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+    implementation Dep.Dagger.core
+    implementation Dep.Dagger.androidSupport
+    kapt Dep.Dagger.compiler
+    kapt Dep.Dagger.androidProcessor
+    compileOnly Dep.Dagger.assistedInjectAnnotations
+    kapt Dep.Dagger.assistedInjectProcessor
+
+    implementation Dep.Groupie.groupie
+    implementation Dep.Groupie.databinding
```
Comment (takahirom) [inline feature/notification/build.gradle]: I think we can remove UI libraries. 🙇 
    implementation Dep.Groupie.groupie
    implementation Dep.Groupie.databinding
    implementation Dep.AndroidX.recyclerView
    implementation Dep.Coil.coil
    implementation Dep.AndroidX.emoji
    implementation Dep.AndroidX.design

**Comment by takahirom** [comment]
We can download a new notification icon from Figma. Can you apply that? 🙇 
https://github.com/DroidKaigi/conference-app-2020#design

File: feature/notification/src/main/AndroidManifest.xml [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+            />
+        <meta-data
+            android:name="com.google.firebase.messaging.default_notification_color"
+            android:resource="@color/color_primary"
+            />
+        <meta-data
+            android:name="com.google.firebase.messaging.default_notification_channel_id"
+            android:value="default_channel"
+            />
+    </application>
```
Comment (takahirom) [inline feature/notification/src/main/AndroidManifest.xml]: Probably we can remove this element 🙏 (It is for announcement)

Comment (Daishin16) [reply feature/notification/src/main/AndroidManifest.xml]: OK. However, I have never created images for each dpi ...

**Comment by takahirom** [comment]
Probably we should add `SessionAlarm` class for showing a notification when my session starts 🙏

**Comment by takahirom** [comment]
Thanks for implementing a lot. 👍 
This is my mistake with the easy label. Please tell me if you want to give up. 🙇

**Comment by Daishin16** [comment]
I will give up because I am unlikely to be able to use this issue anymore It's a good study Also if there is an issue that I can do, thank you at that time

**Comment by takahirom** [comment]
Thank you! We look forward to contributing again!

**Pull request closed without merging**